File: shootout/fib.ml
let rec fib n =
  if n < 2 then 1 else
  fib (n - 1) + fib (n - 2) in
let rec loop n =
  if n = 0 then () else
  (print_int (fib 30);
   print_newline ();
   loop (n - 1)) in
loop 100

File: shootout/fib.c
#include <stdio.h>

int fib(int n) {
  return (n < 2) ? 1 : (fib(n - 1) + fib(n - 2));
}

int main() {
  int i;
  for (i = 0; i < 100; i++) {
    printf("%d\n", fib(30));
  }
  return 0;
}
